lhotse/recipes/librilight.py

```diff
@@ -42,28 +43,40 @@ def _parse_utterance(
     audio_path: Pathlike,
 ) -> Optional[Tuple[Recording, SupervisionSegment]]:
     file_name = str(audio_path).replace(".flac", "").replace(str(corpus_dir) + "/", "")
-    speaker = str(audio_path).split("/")[-3]
     audio_path = audio_path.resolve()
 
     if not audio_path.is_file():
         logging.warning(f"No such file: {audio_path}")
         return None
 
+    audio_info_path = str(audio_path).replace("flac", "json")
+    with open(audio_info_path) as f:
+        audio_infos = json.load(f)
+        speaker = audio_infos["speaker"]
+        vad_infos = audio_infos["voice_activity"]
+
     recording = Recording.from_file(
         path=audio_path,
         recording_id=file_name,
     )
-    segment = SupervisionSegment(
-        id=file_name,
-        recording_id=file_name,
-        start=0.0,
-        duration=recording.duration,
-        channel=0,
-        language="English",
-        speaker=speaker,
-    )
 
-    return recording, segment
+    segments = []
+    segment_seq = 0
+    for vad_info in vad_infos:
+        segments.append(
+            SupervisionSegment(
+                id=file_name + "_" + str(segment_seq),
+                recording_id=file_name,
+                start=vad_info[0],
+                duration=vad_info[1] - vad_info[0],
```

Comment:
I’d suggest using lhotse.utils.add_durations() here to avoid annoying rounding errors from subtraction of two floats.

Reply:
Sure.